Playback 2023: fix hanging loading the episodes

leandroalonso opened the pull request:
| 📘 Part of: #1142 |
|:---:|

There's a bug that can occur if the user has no episodes to sync.

The sync code ends up on `totalEpisodesDispatchGroup.wait()` and it will never leave. That's why I didn't handle the case where there are no episodes to sync. 🤡 

## To test

1. Do a clean install
1. Log in to an account that has a few episodes listened this year and last year
3. Go to Profile and tap the End of Year card
4. ✅ Your data should load correctly

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

leandroalonso committed:
If there are no episodes to sync, leave the total episodes dispatch group

leandroalonso commented on Modules/Server/Sources/PocketCastsServer/Public/Sync/SyncYearListeningHistoryTask.swift:
```diff
@@ -95,6 +95,9 @@ class SyncYearListeningHistoryTask: ApiBaseTask {
                 success = true
+
+                // If there are no episodes to sync we leave the dispatch group
+                totalEpisodesDispatchGroup.leave()
```
There's a way to improve this (unfortunate) collection of dispatch groups. However, I think that for the release branch, we should be careful so I'm making as less changes as possible.

leandroalonso committed:
Only display the logo if not rendering it for share

leandroalonso commented on podcasts/End of Year/Stories/EpilogueStory.swift:
```diff
@@ -77,13 +77,15 @@ struct EpilogueStory: ShareableStory {
+                if !renderForSharing {
+                    VStack {
                         Spacer()
+                        HStack {
```
I also added this since is a small change.

To test:

1. Go to the last story
2. Trigger a screenshot
3. Tap "Share"
4. Save the image
5. ✅ Verify there is just a single logo at the bottom

emilylaguna reviewed (changes requested):
While testing I noticed that if there are multiple year requests, and one of them fails the app will be stuck loading:

To force reproduce this state, I changed SyncYearListeningHistoryTask.swift:69 from

`let url = ServerConstants.Urls.api() + "history/year"` 
to 
`let url = yearToSync == 2023 ? ServerConstants.Urls.api() + "history/year" : "https://localhost/"`

leandroalonso committed:
Simplify SyncYearListeningHistoryTask by moving dispatch group to the outside

leandroalonso commented:
@emilylaguna good catch! I decided to change a bit this code to get rid of all of these `DispatchGroup`. The changes seem a little bit bigger but it's not:

* I reverted `SyncYearListeningHistoryTask` to how it was before handling multiple years. Now it handles just one year;
* I used a `DisaptchGroup` outside of it. Given we've kept the previous behavior, the operation will always succeed or fail
* This cleaned a lot of the mess using the `DispatchGroup`

I've also updated the testing steps.

emilylaguna reviewed (approved):
I tested this with both scenarios and they're both handled now, I also tested the success state, no internet, logged out to purchasing, and all is working. 

I also tested the screenshot detection and no longer see the double PC logo.
